lib/Doctrine/MongoDB/Aggregation/Expr.php

```diff
@@ -584,6 +584,103 @@ public function ifNull($expression, $replacementExpression)
+     * @see https://docs.mongodb.com/manual/reference/operator/aggregation/in/
+     * @since 1.5
+     * @param mixed|self $expression
+     * @param mixed|self $arrayExpression
+     * @return Expr
+     */
+    public function in($expression, $arrayExpression)
+    {
+        return $this->operator('$in', [$expression, $arrayExpression]);
+    }
+
+    /**
+     * Searches an array for an occurence of a specified value and returns the
+     * array index (zero-based) of the first occurence. If the value is not
+     * found, returns -1.
+     *
+     * @see https://docs.mongodb.com/manual/reference/operator/aggregation/indexOfArray/
+     * @since 1.5
+     * @param mixed|self $arrayExpression Can be any valid expression as long as it resolves to an array.
+     * @param mixed|self $searchExpression Can be any valid expression.
+     * @param mixed|self $start Optional. An integer, or a number that can be represented as integers (such as 2.0), that specifies the starting index position for the search. Can be any valid expression that resolves to a non-negative integral number.
+     * @param mixed|self $end An integer, or a number that can be represented as integers (such as 2.0), that specifies the ending index position for the search. Can be any valid expression that resolves to a non-negative integral number.
+     * @return Expr
+     */
+    public function indexOfArray($arrayExpression, $searchExpression, $start = null, $end = null)
+    {
+        $args = [$arrayExpression, $searchExpression];
+        if ($start !== null) {
+            $args[] = $start;
+        }
+        if ($end !== null) {
+            $args[] = $end;
+        }
+
+        return $this->operator('$indexOfArray', $args);
+    }
+
+    /**
+     * Searches a string for an occurence of a substring and returns the UTF-8
+     * byte index (zero-based) of the first occurence. If the substring is not
+     * found, returns -1.
+     *
+     * @see https://docs.mongodb.com/manual/reference/operator/aggregation/indexOfBytes/
+     * @since 1.5
+     * @param mixed|self $stringExpression Can be any valid expression as long as it resolves to a string.
+     * @param mixed|self $substringExpression Can be any valid expression as long as it resolves to a string.
+     * @param int|null $start An integral number that specifies the starting index position for the search. Can be any valid expression that resolves to a non-negative integral number.
+     * @param int|null $end An integral number that specifies the ending index position for the search. Can be any valid expression that resolves to a non-negative integral number.
+     *
+     * @return Expr
+     */
+    public function indexOfBytes($stringExpression, $substringExpression, $start = null, $end = null)
+    {
+        $args = [$stringExpression, $substringExpression];
+        if ($start !== null) {
+            $args[] = $start;
+        }
+        if ($end !== null) {
+            $args[] = $end;
+        }
```

Comment:
Similar issue here if `$start` is null and `$end` is non-null. Looks like there are other cases below, but I'll refrain from spamming.